optimizely/decision_service.py

```diff
@@ -34,20 +35,28 @@ def __init__(self, config, user_profile_service):
     self.config = config
     self.logger = config.logger
 
-  @staticmethod
-  def _get_bucketing_id(user_id, attributes):
+  def _get_bucketing_id(self, user_id, attributes):
     """ Helper method to determine bucketing ID for the user.
 
     Args:
       user_id: ID for user.
       attributes: Dict representing user attributes. May consist of bucketing ID to be used.
 
     Returns:
-      String representing bucketing ID for the user. Fallback to user's ID if not provided.
+      String representing bucketing Id if it is a String type in attributes else return userId.
     """
 
     attributes = attributes or {}
-    return attributes.get(enums.ControlAttributes.BUCKETING_ID, user_id)
+    bucketing_id = attributes.get(enums.ControlAttributes.BUCKETING_ID)
+
+    if bucketing_id is not None:
+      if isinstance(bucketing_id, string_types):
+        self.logger.debug('Bucketing ID is valid: "%s".' % (bucketing_id))
```

Comment:
I think this message is not useful as it will happen often that bucketing ID is fine and it is a lot of new unnecessary messageds. I'd drop this and just use the warning message.